### PR TITLE
bonus Python typing

### PR DESCRIPTION
**Pull request by mr-c**
## How to test the changes?
(Select all options that apply)
- [ ] I've included appropriate [automated tests](https://docs.galaxyproject.org/en/latest/dev/writing_tests.html).
- [x] This is a refactoring of components with existing test coverage.
- [ ] Instructions for manual testing are as follows:
  1. [add testing steps and prerequisites here if you didn't write automated tests covering all your changes]

## License
- [x] I agree to license these and all my past contributions to the core galaxy codebase under the [MIT license](https://opensource.org/licenses/MIT).

### CODE REVIEW

File: lib/galaxy/tool_util/data/__init__.py [hunk 1]
```diff
@@ -109,20 +113,20 @@ def from_dict(cls, d):
...
         self,
-        config_element,
-        tool_data_path,
-        from_shed_config=False,
-        filename=None,
-        tool_data_path_files=None,
-        other_config_dict=None,
-    ):
+        config_element: _Element,
+        tool_data_path: PosixPath,
```
Comment (nsoranzo) [inline lib/galaxy/tool_util/data/__init__.py]:
```suggestion
        tool_data_path: Union[os.PathLike, str],
```

File: lib/galaxy/tool_util/data/__init__.py [hunk 2]
```diff
@@ -109,20 +113,20 @@ def from_dict(cls, d):
...
+        from_shed_config: bool = False,
+        filename: Optional[PosixPath] = None,
```
Comment (nsoranzo) [inline lib/galaxy/tool_util/data/__init__.py]:
```suggestion
        filename: Optional[Union[os.PathLike, str]] = None,
```

File: lib/galaxy/tool_util/data/__init__.py [hunk 3]
```diff
@@ -214,13 +218,13 @@ class TabularToolDataTable(ToolDataTable):
...
         self,
-        config_element,
-        tool_data_path,
-        from_shed_config=False,
-        filename=None,
-        tool_data_path_files=None,
-        other_config_dict=None,
-    ):
+        config_element: _Element,
+        tool_data_path: PosixPath,
```
Comment (nsoranzo) [inline lib/galaxy/tool_util/data/__init__.py]:
```suggestion
        tool_data_path: Union[os.PathLike, str],
```

File: lib/galaxy/tool_util/data/__init__.py [hunk 4]
```diff
@@ -214,13 +218,13 @@ class TabularToolDataTable(ToolDataTable):
...
+        from_shed_config: bool = False,
+        filename: Optional[PosixPath] = None,
```
Comment (nsoranzo) [inline lib/galaxy/tool_util/data/__init__.py]:
```suggestion
        filename: Optional[Union[os.PathLike, str]] = None,
```

File: lib/galaxy/tool_util/data/__init__.py [hunk 5]
```diff
@@ -233,7 +237,9 @@ def __init__(
         self.data = []
         self.configure_and_load(config_element, tool_data_path, from_shed_config)
 
-    def configure_and_load(self, config_element, tool_data_path, from_shed_config=False, url_timeout=10):
+    def configure_and_load(
+        self, config_element: _Element, tool_data_path: PosixPath, from_shed_config: bool = False, url_timeout: int = 10
```
Comment (nsoranzo) [inline lib/galaxy/tool_util/data/__init__.py]:
```suggestion
        self, config_element: _Element, tool_data_path: Union[os.PathLike, str], from_shed_config: bool = False, url_timeout: int = 10
```

**Comment by mr-c** [comment]
Thank you @nsoranzo !

File: test/unit/webapps/test_routes.py [hunk 6]
```diff
@@ -22,7 +25,7 @@ def assert_maps(self, url, method="GET", **parts):
 
 def test_galaxy_routes():
     test_config = Bunch(template_cache_path="/tmp")
-    app = Bunch(config=test_config, security=object(), trace_logger=None, name="galaxy")
+    app = cast(MinimalApp, Bunch(config=test_config, security=object(), trace_logger=None, name="galaxy"))
```
Comment (jmchilton) [inline test/unit/webapps/test_routes.py:28]: We shouldn't do this - it isn't a that thing.

**Pull request merged**